Add DOI badge to subdataset view

mslw opened the pull request:
A DOI badge will be shown in the subdataset card if a DOI is available. Uses https://badgen.net/

Badgen was chosen over shields.io mostly because it is slightly easier to format the URL (using encodeURIComponent instead of replacing hyphens and underscores). It also promises to be faster and have a leaner code underneath.

A future rework could implement a simplified shield representation using divs with backgrounds (similar to what GIN does) or include the badgen JS library (permissive license) to remove reliance on external service.

These are the current visuals, showing a card with and without DOI (note: the second dataset also has one, just not yet in the catalog).

![image](https://github.com/sfb1451/metadata-catalog/assets/11985212/bbff26e9-d2e4-4ee2-b0f9-cadad4340314)


Closes #22
Closes #101

TODO:
- [x] This would work best if combined with re-extraction of some GIN datasets, taking advantage from the ability of looking up the DOI on GIN website. As it stands, we have 4 DOIs, two leading to actual datasets, one kind-of (leads to publication but the dataset is supplementary data in this case) and one to the publication; #101 
- [ ] Consider removing the external dependency -- maybe in the future

mslw committed:
Add DOI badge to subdataset view

A DOI badge will be shown in the subdataset card if a DOI is
available. Uses https://badgen.net/

Badgen was chosen over shields.io mostly because it is slightly easier
to format the URL (using encodeURICOmponent instead of replacing
hyphens and underscores). It also promises to be faster and leaner.

A future rework could implement a simplified shield representation
using divs with backgrounds (similar to what GIN does) or include the
badgen JS library (permissive license) to remove reliance on external
service.

Closes #22

mih commented:
I like the look

jsheunis commented:
Looks good, nice placement 👍 I wasn't aware of badgen, thanks for introducing it. I agree, simplest is probably to use local css to create a custom badge, but the current solution works and is easy enough.

mslw committed:
Add missing DOIs

This is a mostly-manual addition of metadata for four datasets; three
of them have GIN DOIs (which should be retrieved next time the whole
dataset is reextracted, here it was not necesseary) and one of them
has a Jülich Data DOI which is added purely by hand here.
